### PR TITLE
Digital post

### PR DESCRIPTION
**Pull request by rimi-itk**
Adds Digital post (`os2forms_digital_post`) module (cf. [modules/os2forms_digital_post/README.md](https://github.com/OS2Forms/os2forms/blob/730539726db9c52519759cb9c4f82c170ad29a9d/modules/os2forms_digital_post/README.md)).

This is a follow-up to https://github.com/OS2Forms/os2forms/pull/56 where we had some workarounds to handle the case when `os2forms_digital_post` was not installed or available.

**Note**: The digital post module have two dependencies† that currently live in the `itk-dev` vendor namespace and to people in the know this may seem a bit odd. These dependencies, however, make sense outside OS2Forms and are therefore _not moved_ into https://github.com/OS2Forms/os2forms. It would make sense to move them to a general OS2 organization, e.g. https://github.com/os2web, say, but we don't yet know how to actually handle this (it's a question of resources; finances, responsibility, governance etc.).

**Another note**: Our/Drupal's coding standards do not require `@param` annotations on all function parameters and therefore these are not added on all functions, but only where they add real value. 

† a Drupal module: https://github.com/itk-dev/beskedfordeler-drupal and a library: https://github.com/itk-dev/serviceplatformen

### CODE REVIEW

**Review by stanbellcom — APPROVED** [review]

File: modules/os2forms_digital_post/CHANGELOG.md
[the remark this answers is not in the document]
Comment (rimi-itk) [reply modules/os2forms_digital_post/CHANGELOG.md:1]: You're absolutely right. Removed in https://github.com/OS2Forms/os2forms/pull/62/commits/29eae58decde8494ccc81f185ab84e32e63a2707.

File: modules/os2forms_digital_post/os2forms_digital_post.install [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+function os2forms_digital_post_schema() {
+  return Drupal::service(BeskedfordelerHelper::class)->schema();
+}
+
+/**
+ * Enable Beskedfordeler module.
+ */
+function os2forms_digital_post_update_9001(): void {
+  Drupal::service('module_installer')->install(['beskedfordeler']);
+}
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply modules/os2forms_digital_post/os2forms_digital_post.install]: You're absolutely right. Removed in https://github.com/OS2Forms/os2forms/pull/62/commits/29eae58decde8494ccc81f185ab84e32e63a2707.

We've been using the `os2forms_digital_post` module for a while on selvbetjening.aarhuskommune.dk and this was a leftover from that journey.

File: modules/os2forms_digital_post/os2forms_digital_post.install [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Create beskedfordeler table.
+ */
+function os2forms_digital_post_update_9002(): void {
+  Drupal::service(BeskedfordelerHelper::class)->update9002();
+}
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply modules/os2forms_digital_post/os2forms_digital_post.install]: You're absolutely right. Removed in https://github.com/OS2Forms/os2forms/pull/62/commits/29eae58decde8494ccc81f185ab84e32e63a2707.

File: modules/os2forms_digital_post/phpstan.neon
[the remark this answers is not in the document]
Comment (rimi-itk) [reply modules/os2forms_digital_post/phpstan.neon:1]: You're absolutely right. Removed in https://github.com/OS2Forms/os2forms/pull/62/commits/29eae58decde8494ccc81f185ab84e32e63a2707.

File: modules/os2forms_digital_post/src/Helper/BeskedfordelerHelper.php [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      ],
+    ];
+  }
+
+  /**
+   * Implements hook_update_N().
+   *
+   * Creates beskedfordeler table.
+   */
+  public function update9002(): void {
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply modules/os2forms_digital_post/src/Helper/BeskedfordelerHelper.php]: Removed in https://github.com/OS2Forms/os2forms/pull/62/commits/29eae58decde8494ccc81f185ab84e32e63a2707.

File: modules/os2forms_digital_post/src/Model/Message.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      'message_uuid' => 'messageUUID',
+      'beskedfordeler_message' => 'beskedfordelerMessage',
+    ][$name] ?? $name;
+
+    if (!property_exists($this, $property)) {
+      throw new \RuntimeException(sprintf('Invalid property: %s', $property));
+    }
+
+    $this->$property = $value;
+  }
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply modules/os2forms_digital_post/src/Model/Message.php:70]: Took me a while to figure it out too, but at all makes sense. I've added some comments in https://github.com/OS2Forms/os2forms/pull/62/commits/52a0d0c2eae2459cce4deefb1a52597359beaab2.

File: modules/os2forms_digital_post/phpcs.xml.dist
[the remark this answers is not in the document]
Comment (rimi-itk) [reply modules/os2forms_digital_post/phpcs.xml.dist:1]: You're absolutely right. Removed in https://github.com/OS2Forms/os2forms/pull/62/commits/29eae58decde8494ccc81f185ab84e32e63a2707.

**Comment by rimi-itk** [comment]
@stanbellcom, I've cleaned up and added some comments in the code. Thanks for useful feedback.

**Review by stanbellcom — APPROVED** [review]

**Pull request merged**